src/qibolab/dummy/platform.py

```diff
@@ -11,27 +11,39 @@
 def create_dummy() -> Platform:
     """Create a dummy platform using the dummy instrument."""
     # register the instruments
-    instrument = DummyInstrument("dummy", "0.0.0.0")
-    pump = DummyLocalOscillator("twpa_pump", "0.0.0.0")
+    instrument = DummyInstrument(name="dummy", address="0.0.0.0")
+    pump = DummyLocalOscillator(name="twpa_pump", address="0.0.0.0")
 
     qubits = {}
     # attach the channels
     for q in range(5):
-        probe, acquisition = f"qubit_{q}/probe", f"qubit_{q}/acquisition"
+        drive, drive12, flux, probe, acquisition = (
+            f"qubit_{q}/drive",
+            f"qubit_{q}/drive12",
+            f"qubit_{q}/flux",
+            f"qubit_{q}/probe",
+            f"qubit_{q}/acquisition",
+        )
+        instrument.channels |= {
+            probe: IqChannel(mixer=None, lo=None),
+            acquisition: AcquireChannel(twpa_pump=pump.name, probe=probe),
+            drive: IqChannel(mixer=None, lo=None),
+            drive12: IqChannel(mixer=None, lo=None),
+            flux: DcChannel(),
+        }
```

Comment:
Ah, yes, it's possible. I've just been lazy, and kept the change minimal (the instrument was instantiated before, and it is happening in most cases - but it doesn't have to be).

In any case, instruments are not supposed to be serialized, and initially I wasn't planning to migrate to Pydantic, but keeping them dataclasses. In practice, I did it for the LOs and the abstract, because Pydantic is also more robust wrt subclassing (e.g. it doesn't have the limitation of having defaults at the end, so you may define required positional arguments, without any default, even if the parent class is already defining defaults). So, I didn't try to be too strict with the constraint we're imposing on the serialized classes, i.e. I didn't bother freezing. But we can, of course.

Reply:
Done in a922375